web/auth/session.go

```diff
@@ -141,17 +137,44 @@ func (s *Session) Delete(i *instance.Instance) *http.Cookie {
 }
 
 // ToCookie returns an http.Cookie for this Session
-// TODO SECURITY figure out if we keep session in couchdb or not
-// if we do, check if ID is random enough on the whole server to use as a
-// sessionid
-func (s *Session) ToCookie() *http.Cookie {
+func (s *Session) ToCookie() (*http.Cookie, error) {
+	encoded, err := crypto.EncodeAuthMessage(cookieMACConfig(s.Instance), s.ID())
+	if err != nil {
+		return nil, err
+	}
+
 	return &http.Cookie{
 		Name:     SessionCookieName,
-		Value:    s.ID(),
+		Value:    string(encoded),
 		MaxAge:   SessionMaxAge,
 		Path:     "/",
 		Domain:   "." + s.Instance.Domain,
 		Secure:   true,
 		HttpOnly: true,
+	}, nil
+}
+
+// cookieMACConfig returns the options to authenticate the session cookie.
+//
+// We rely on a MACed cookie value, without additional encryption of the
+// message since it should not contain critical private informations and is
+// protected by HTTPs (secure cookie).
+//
+// About MaxLength, for a session of size 100 bytes
+//
+//       8 bytes time
+//   +  32 bytes HMAC-SHA256
+//   + 100 bytes session
+//   + base64 encoding (4*n/3 + 2 padding)
+//   < 200 bytes
+//
+// 2048 bytes should be sufficient enough to support any type of session.
```

Comment:
IMHO, 2048 is too much. A TCP packet is less than 1500 bytes, and cookies are sent often (less with http2, but still). I think we should limit to 256 or 512 bytes.